crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -62,23 +65,57 @@ pub(crate) async fn farm_multi_disk(
         .map(|single_disk_plot| single_disk_plot.wait())
         .collect::<FuturesUnordered<_>>();
 
-    select(
-        Box::pin(async move {
+    futures::select!(
+        // Signal future
+        _ = Box::pin(async move {
             signal.await;
+        }).fuse() => {},
 
-            Ok(())
-        }),
-        Box::pin(async move {
+        // Plotting future
+        _ = Box::pin(async move {
             while let Some(result) = single_disk_plots_stream.next().await {
                 result?;
 
                 info!("Farm exited successfully");
             }
-
             anyhow::Ok(())
-        }),
-    )
-    .await
-    .factor_first()
-    .0
+        }).fuse() => {},
+
+        // Node runner future
+        _ = Box::pin(async move {
+            if let Some(mut node_runner) = node_runner{
+                node_runner.run().await;
+
+                info!("Node runner exited.")
+            } else {
+                futures::future::pending().await
+            }
+        }).fuse() => {},
+    );
+
+    anyhow::Ok(())
+}
+
+async fn configure_dsn(
+    enable_dsn: bool,
+    listen_on: Vec<Multiaddr>,
+    bootstrap_nodes: Vec<Multiaddr>,
+) -> Result<(Option<Node>, Option<NodeRunner>), anyhow::Error> {
+    if !enable_dsn {
+        info!("No DSN configured.");
+        return Ok((None, None));
+    }
```

Comment:
I would create it in all cases, but only used when the flag is set. Makes code a lot simpler.

Reply:
I used Option<Node> further as an indicator that DSN is enabled instead of passing an additional flag.

Reply:
You can still use `Option` in the piece getter